src/report_reader.cpp

```diff
@@ -349,22 +349,22 @@ DataFrame<T> ReportReader<T>::Population::get(const nonstd::optional<Selection>&
         node_ids = selection->flatten();
```

Comment:
why?

Reply:
Using `reserve` we will avoid the memory re-allocation for the vector `node_ids` when it's populated.
Since we know the size of the final vector, we could do this before the `transform` loop.

Reply:
But here, the function produce a vector of the right size and as this is a prvalue, it will be moved.

Reply:
Maybe I don't understand well because your link seems broken.

Reply:
The link is to the (current) line 340 `std::transform(nodes_pointers_.begin(),`.
Sorry but github does not allow me to comment on that line because it was not modified in this PR.

Reply:
Thanks @GianlucaFicarelli, we just added your change!